### PR TITLE
[IMP] support for sub-themes

### PR DESCRIPTION
**Pull request by yelizariev**
closes #409

### CODE REVIEW

**Comment by yelizariev** [comment]
No time for relaxing @Yajo @pedrobaeza

File: website_multi_theme/models/website.py [hunk 1]
```diff
@@ -183,23 +187,26 @@ def _multi_theme_activate(self):
...
-                    data = etree.fromstring(copied_view.arch)
-                    data.attrib["inherit_id"] = new_parent.key
-                    copied_view.arch = etree.tostring(data)
+                    copied_view._replace_parent(new_parent)
 
                 custom_views |= copied_view
             # Delete any custom view that should exist no more
-            (website.multi_theme_view_ids - custom_views).unlink()
+            views_to_remove = website.multi_theme_view_ids - custom_views
+            # Removed views can be copied parent for someone.
```
Comment (yajo) [inline website_multi_theme/models/website.py]: "Removed views could be a copied parent for others"

File: website_multi_theme/data/themes_default.xml [hunk 2]
```diff
@@ -8,14 +8,8 @@
...
-    <template id="theme_default_dummy" name="Dummy asset for Default Theme">
-        <!-- It's empty, but we have to have one, because otherwise multi-theme
-             magic is not applied for themes without any views -->
-    </template>
-
-
-    <record model="website.theme.asset" id="theme_default_asset">
-        <field name="name">website_multi_theme.theme_default_dummy</field>
+    <record model="website.theme.asset" id="theme_default_asset_footer_custom">
+        <field name="name">website.footer_custom</field>
```
Comment (yajo) [inline website_multi_theme/data/themes_default.xml:12]: Shouldn't we include the header by default also?

**Review by yajo — COMMENTED** [review]
Ooook... I see the code and I think I sorta understand it somehow... but TBH I need you to draw here the big picture on what, why and how this. Could you please? :innocent:

File: website_multi_theme/data/themes_default.xml (hunk 2 above)
Comment (yelizariev) [reply website_multi_theme/data/themes_default.xml:12]: Which header do you mean? Could you say xmlid of the view

**Comment by yelizariev** [comment]
So, the big picture is:

we copy for each website not just views of one theme (module), but views of other themes ("dependencies"). Those themes are not necessarily technical dependencies of the main module. Moreover, a theme might be not a module at all (e.g. "Default Theme").

This updates along with previous (merged) updates also allow converting not just a theme to multi-theme view, but a specific view to multi-website view (as it's done with ``website.custom_footer``). I'll add this to README

**Comment by yelizariev** [comment]
I've updated the README with a hope that it brings more developers that understand what is going on here :smiley_cat:

File: website_multi_theme/data/themes_default.xml (hunk 2 above)
Comment (yajo) [reply website_multi_theme/data/themes_default.xml:12]: You're right, it's not a header. it was `website.layout`.

**Review by yajo — APPROVED** [review]
OK I think I got it now.

File: website_multi_theme/README.rst [hunk 3]
```diff
@@ -8,6 +8,37 @@ Website Multi Theme
 
 Allow the website admin to set a different theme for each website.
 
+The *theme* might be not just a theme-module, but any set of themes and even
+particual views from any module (e.g. view ``website.custom_footer`` from
```
Comment (pedrobaeza) [inline website_multi_theme/README.rst]: s/particual/particular

**Review by pedrobaeza — APPROVED** [review]
Little nitpicking

**Comment by yelizariev** [comment]
Spelling is fixed. The PR is green

**Pull request merged**